### PR TITLE
Migrate image from centos:7 to almalinux:9

### PR DESCRIPTION
**Pull request by mrguilima**
First version of migrating the Dockerfile from celtos:7 to almalinux:9

### CODE REVIEW

File: wm/docker/image/Dockerfile [hunk 1]
```diff
@@ -1,77 +1,80 @@
...
+RUN dnf install -y libaio\
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
+
+#RUN rpm -i https://download.oracle.com/otn_software/linux/instantclient/2112000/el9/oracle-instantclient6-basic-21.12.0.0.0-1.el9.x86_64.rpm
+#RUN dnf install -y oracle-release-el9
+#RUN dnf install -y oracle-instantclient-release-el9
+#RUN dnf install -y oracle-instantclient-basic
+COPY oracle-instantclient-basic-21.12.0.0.0-1.el9.x86_64.rpm /tmp
+RUN dnf install -y /tmp/oracle-instantclient-basic-21.12.0.0.0-1.el9.x86_64.rpm
```
Comment (dynamic-entropy) [inline wm/docker/image/Dockerfile]: This assumes that we already downloaded the rpm. 

I believe one option is to add the curl command to download the rpm; then install it; and then remove the rpm. 
Its 72 MB; so it should be fine. 

The second option is to add the oracle yum server to the package manger configuration. 
See instructions [here](https://yum.oracle.com/getting-started.html#:~:text=Red%20Hat%20Enterprise%20Linux%209)

Comment (dynamic-entropy) [reply wm/docker/image/Dockerfile]: @ericvaandering any preferences, here?

**Review by dynamic-entropy — CHANGES_REQUESTED** [review]
Btw, I think you are working on the wrong dockerfile. 
This is the main one, 
https://github.com/dmwm/cms_consistency/blob/master/docker/Dockerfile

While exploring the repo, I felt like a bit of cleaning up and organisation would help. We would do that as we develop our understanding of what is what.

**Comment by ericvaandering** [comment]
Ready to merge? @mrguilima does this build and push to the registry?

**Comment by mrguilima** [comment]
I have updated build_image.sh with CONSISTENCY_VERSION=4.5.1. (from 4.5.0).  Is this okay?
This build_image.sh is run automatically?  The last line seems to do the pushing, so I think it will build and push to the registry.

After my local image building tests, I used "podman run -it" to run it locally, and from the prompt I tried to launch run.sh, it fails with:
```/config/config.yaml not found```
There are several tests on top of the run.sh script, pointing to files which are missing from the image  -- are those volumes mounted during the Kubernetes configuration?

**Comment by dynamic-entropy** [comment]
Hello @mrguilima 
To answer some of your questions (Except what should be the version increment if any, that I leave to @ericvaandering to clarify):
1. We need to run the build.sh ourselves; 
 - We also have a GitHub action (https://github.com/dmwm/cms_consistency/blob/master/.github/workflows/release-images.yml) that will build and push it for you. It does not rely on the `build.sh` script though. It works based on tags (same as cmsrucio repo). 
@ericvaandering have you already added the secrets to this repo? https://github.com/dmwm/cms_consistency/pull/2#issuecomment-2060854386

2. Yes, these config files are mounted in the containers as follows, https://github.com/dmwm/CMSRucio/blob/c043e76433260e3e38690bbf0f98bef272b93db3/helm/rucio-consistency/templates/deployment.yaml#L67-L68

**Comment by dynamic-entropy** [comment]
One question I have for @ericvaandering  is; do we or anyone else even use the image in the `wm` folder?
https://github.com/dmwm/cms_consistency/blob/master/wm/docker/image/Dockerfile

In fact, is anything in that entire folder used? If yes, by whom?

**Comment by dynamic-entropy** [comment]
This is merged. But it will never be built by the CI because it requires the existence of Oracle client in the build environment, https://github.com/dmwm/cms_consistency/pull/3#discussion_r1595075945

Can this be fixed, please?

@ericvaandering @mrguilima

**Comment by mrguilima** [comment]
I thought this was installed.  I specifically found an URL to be used for the installation.  I am checking.

**Comment by dynamic-entropy** [comment]
@mrguilima 
I see, https://github.com/dmwm/cms_consistency/blob/76e7026a1918ac5fe894da1122defb8d25765128/docker/Dockerfile#L35

In that case, we only need to add Stephans's suggestions to the docker file.
And then, can you try building the image by pushing a tag to this repo?

**Comment by mrguilima** [comment]
Sorry @dynamic-entropy, I don't know what you meant with Stephan's suggestions.  Can you provide more information on that, please?  
The current tag is registry.cern.ch/cms-rucio/rucio-consistency:release-4.5.1.  Do you have any suggestion/requirement for the new tag?  Or should I just bump it up to 4.5.2?

**Comment by mrguilima** [comment]
Oh, I see Stephan's suggestion on the Mattermost channel.  Working on the dockerfile part, who is going to do the pupper part?

**Comment by mrguilima** [comment]
Please approve the 4.5.2 tag, or suggest a new one.

**Comment by dynamic-entropy** [comment]
We don't use puppet etc. 
Just the docker file needs to be updated. 
Yup, 4.5.2 is good. 

For the Github Actions CI, either @ericvaandering will have to add it to the repo or give us permission to push tags.

**Pull request merged**